scribejava-core/src/main/java/com/github/scribejava/core/model/OAuth2ErrorResponse.java

```diff
@@ -0,0 +1,36 @@
+package com.github.scribejava.core.model;
+
+import java.net.URI;
+
+// https://tools.ietf.org/html/rfc6749#section-5.2
+public class OAuth2ErrorResponse {
+
+    public enum OAuthError {
+        invalid_request, invalid_client, invalid_grant, unauthorized_client, unsupported_grant_type, invalid_scope
+    }
+
+    private final OAuthError error;
+    private final String errorDescription;
+    private final URI errorUri;
```

Comment:
OAuth2ErrorResponse(OAuthError error, String errorDescription, URI errorUri, String rawResponse)

and the rawResponse can be null in some cases